### ISSUE TITLE
springdoc.writer-with-order-by-keys could be more deterministic

### ISSUE THREAD

**Issue by ianwallen** [springdoc/springdoc-openapi#1362]
**Is your feature request related to a problem? Please describe.**
The following option is not as deterministic as expected. (Not sure if this is bug or a feature request)
`springdoc.writer-with-order-by-keys=true`

Reversing function order of the methods can cause different documents to be generated so the results are not really deterministic. The order of the paths seems to remain ordered, but the other attributes inside are not.   This makes it difficult to perform a comparison between two open api documents.

**Describe the solution you'd like**
- I would like to see all the attributes orders the same way so that it makes a deterministic document that is easy to compare to an older version.

**Describe alternatives you've considered**
- Maybe the documentation could be more clear that it is only ordering the "paths" or maybe remove deterministic from the documentation.
Current documentation states `Enable a deterministic/alphabetical ordering.`

**Additional context**
- To show the problem edit the following unit test.

https://github.com/springdoc/springdoc-openapi/blob/0fb9ee43de0c4a3ff29069bb37b49c72de94908e/springdoc-openapi-webmvc-core/src/test/java/test/org/springdoc/api/app138/HelloController.java#L25-L34

And reverse the methods so that they look like the following

```java
public class HelloController {

	@GetMapping("/testB")
	public void testB(String hello) {
	}

	@GetMapping("/testA")
	public void testA(String hello) {
	}
}
```

This will cause the unit test to fail with the following error.

```
Expected :{"openapi":"3.0.1","info":{"title":"OpenAPI definition","version":"v0"},"servers":[{"url":"http://localhost","description":"Generated server url"}],"paths":{"/testA":{"get":{"tags":["hello-controller"],"operationId":"testA","parameters":[{"name":"hello","in":"query","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"OK"}}}},"/testB":{"get":{"tags":["hello-controller"],"operationId":"testB","parameters":[{"name":"hello","in":"query","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"OK"}}}}},"components":{}}
Actual   :{"openapi":"3.0.1","info":{"title":"OpenAPI definition","version":"v0"},"servers":[{"url":"http://localhost","description":"Generated server url"}],"paths":{"/testA":{"get":{"operationId":"testA","parameters":[{"in":"query","name":"hello","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"OK"}},"tags":["hello-controller"]}},"/testB":{"get":{"operationId":"testB","parameters":[{"in":"query","name":"hello","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"OK"}},"tags":["hello-controller"]}}},"components":{}}
```

We can see that the Paths are in the correct order but the attributes such as "tags", "operationId" have changed order.

**Comment by ianwallen** [comment]
Looks like the issue may be the following

https://github.com/springdoc/springdoc-openapi/blob/0fb9ee43de0c4a3ff29069bb37b49c72de94908e/springdoc-openapi-common/src/main/java/org/springdoc/api/AbstractOpenApiResource.java#L1276-L1281

The following is a good read that may help identify some of the issues

https://www.stubbornjava.com/posts/creating-a-somewhat-deterministic-jackson-objectmapper

**Comment by bnasslahsen** [comment]
@ianwallen, 

Even after inverting the order of method, no issue identified using CI as you can see.
I am closing this ticket and it can be reopened if any relevant information are provided. 
Feel free to update the ticket, if you are able to provide a **Minimal, Reproducible Example - with HelloController**.

**Comment by ianwallen** [comment]
This is odd? When I run all the test, they all  pass including test 138.  When I only run test 138 as a single test then the test fails?  So it seems to be affected by the other tests?

I was running it as a single test using Intellij and it was failing.  But I have seen odd behavior in the past in the way that Intellij runs the tests so I ran maven commands directly instead and I got the same results.

I did the following to generate the failure using maven commands.

```
mvn clean install
cd springdoc-openapi-webmvc-core
mvn test -Dtest=test.org.springdoc.api.app138.SpringDocApp138Test -f pom.xml
```
